### DISCUSSION TITLE
vNet with multiple address spaces

### DISCUSSION THREAD

**Discussion by vladimirs-davidovs** [Azure/ipam#117] [Q&A] [answered]
Hello everyone!
We have the following scenario - we need to create a vNet with multiple address spaces, which come from multiple different IPAM reservations. In that case we shall have multiple different 'X-IPAM-RES-ID' values - one per reservation, is there a way how we can tell IPAM, that those reservations were consumed by a single vNet? 
And please forgive me, if this is covered in documentation part or different discussion, that I might have missed.

**Comment by DCMattyG** [comment, marked as answer]
Hi @vladimirs-davidovs, thank you for reaching out to the community!

You are not alone in this request, it has been brought up a number of times before in #94 and #116. The quick answer is that the current process doesn't work well today for Virtual Networks that are created (net-new) with multiple CIDR's. There is a work-around that I can suggest whilst we solve this issue in the product.

You will see in #94, the current workflow they've used is as follows:
1. Create X number of reservations to obtain the CIDR's destined for the new vNET
2. Create the new VNET in Azure with the CIDR's from step 1
3. Execute a PUT against the Block(s) to associate the new vNET Address Spaces
4. Execute a DELETE against the Reservation(s) that were opened in step 1

We fully acknowledge this is not ideal, and are working on a long-term fix. Hopefully this will help for now, but we'll update this discussion (and the attached issue/discussion) once we have something in our DEV branch for users to test and provide some feedback on. Hopefully we can have this done relatively soon (next week or two).

Please let us know if you need anything else in the meantime!

**Reply by vladimirs-davidovs** [reply]
Thank you very much for quick response @DCMattyG! 
After looking at the API, I had the same idea, but wanted to confirm that I was not considering doing something that did not make any sense.

**Reply by DCMattyG** [reply]
You were on the right track! We'll let you know as soon as we add in more "native" support for multiple Address Spaces assigned to a new vNET.

**Reply by gitfrs** [reply]
Hello, we are also very interested in this feature.
I'm currently testing using the Terraform Provider created by xtratusipam
Do you have a timeline for when it would integrated ?
Thanks a lot !

**Reply by DCMattyG** [reply]
Quick update @vladimirs-davidovs and @gitfrs, @XtratusCloud has completed the GA version of the Terraform provider.

More docs can be found here:
https://registry.terraform.io/providers/XtratusCloud/azureipam/latest/docs

I'll work on adding some companion documentation to our official Azure IPAM docs page as well. Hope it works well for you in the meantime!

To answer your question directly @gitfrs, probably sometime in the next 30-45 days. I have a few other major features I'm working on in parallel, but this one is most definitely on my list.

**Comment by DCMattyG** [comment]
Hi @vladimirs-davidovs, in case you hadn't already seen my comments on other discussions & issues, you can now apply a comma-separated list of Reservation ID's on the X-IPAM-RES-ID Tag to cover multiple Address Spaces on a Virtual Network. You can also use Reservations for vWAN Hubs too. Hope this helps you out!

**Reply by vladimirs-davidovs** [reply]
Awesome news! Thank you very much @DCMattyG!

**Reply by vladimirs-davidovs** [reply]
@DCMattyG, I am planning on updating IPAM to the latest version, but before doing that I wanted to ask if there were any breaking changes on the API side, that might affect the code, calling those?

**Reply by DCMattyG** [reply]
No breaking changes @vladimirs-davidovs , you're all clear 👍
